apollo-client-config-data/src/main/java/com/ctrip/framework/apollo/config/data/extension/messaging/ApolloClientExtensionMessagingFactory.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.config.data.extension.messaging;
+
+import com.ctrip.framework.apollo.config.data.extension.enums.ApolloClientMessagingType;
+import com.ctrip.framework.apollo.config.data.extension.properties.ApolloClientExtensionProperties;
+import com.ctrip.framework.apollo.config.data.extension.properties.ApolloClientProperties;
+import com.ctrip.framework.apollo.config.data.extension.webclient.ApolloClientLongPollingMessagingFactory;
+import com.ctrip.framework.apollo.config.data.extension.websocket.ApolloClientWebsocketMessagingFactory;
+import com.ctrip.framework.apollo.config.data.util.Slf4jLogMessageFormatter;
+import org.apache.commons.logging.Log;
+import org.springframework.boot.ConfigurableBootstrapContext;
+import org.springframework.boot.context.properties.bind.BindHandler;
+import org.springframework.boot.context.properties.bind.Binder;
+
+/**
+ * @author vdisk <vdisk@foxmail.com>
+ */
+public class ApolloClientExtensionMessagingFactory {
+
+  private final Log log;
+
+  private final ApolloClientPropertiesFactory apolloClientPropertiesFactory;
+
+  private final ApolloClientLongPollingMessagingFactory apolloClientLongPollingMessagingFactory;
+
+  private final ApolloClientWebsocketMessagingFactory apolloClientWebsocketMessagingFactory;
+
+  public ApolloClientExtensionMessagingFactory(Log log,
+      ConfigurableBootstrapContext bootstrapContext) {
+    this.log = log;
+    this.apolloClientPropertiesFactory = new ApolloClientPropertiesFactory();
+    this.apolloClientLongPollingMessagingFactory = new ApolloClientLongPollingMessagingFactory(log,
+        bootstrapContext);
+    this.apolloClientWebsocketMessagingFactory = new ApolloClientWebsocketMessagingFactory(log,
+        bootstrapContext);
+  }
+
+  /**
+   * prepare extension messaging
+   *
+   * @param binder      properties binder
+   * @param bindHandler properties bind handler
+   */
+  public void prepareMessaging(Binder binder, BindHandler bindHandler) {
```

Comment:
what does messaging do? I didn't see any documentation on this topic

Reply:
> what does messaging do? I didn't see any documentation on this topic

rename the method to `com.ctrip.framework.apollo.config.data.extension.initialize.ApolloClientExtensionInitializeFactory#initializeExtension`